cinema/views.py

```diff
@@ -43,6 +46,26 @@ def get_serializer_class(self):
 
         return MovieSerializer
 
+    def get_queryset(self):
+        queryset = self.queryset
+        if self.action in ("list", "retrieve"):
+            queryset = queryset.prefetch_related("genres", "actors")
+
+        title = self.request.query_params.get("title")
+        genres = self.request.query_params.get("genres")
+        actors = self.request.query_params.get("actors")
+
+        if title:
+            queryset = queryset.filter(title__icontains=title)
+        if genres:
+            genres = [int(str_id) for str_id in genres.split(",")]
```

Comment:
Consider adding error handling for the conversion of genre IDs to integers. If the query string contains non-integer values, it will raise a `ValueError`. You can use a try-except block to handle such cases gracefully.